src/main/java/maksym/fedorenko/bookstore/model/User.java

```diff
@@ -55,6 +60,11 @@ public class User implements UserDetails {
     )
     private Set<Role> roles = new HashSet<>();
 
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @OneToOne(mappedBy = "user", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
```

Comment:
should you specify `orphanRemoveal` since cart cannot exist without user?

btw, read more about [cascade-types](https://vladmihalcea.com/a-beginners-guide-to-jpa-and-hibernate-cascade-types/)